### PR TITLE
AP_ServoRelayEvents: fix do_set_servo

### PR DESCRIPTION
**Pull request by tridge**
this was broken by the recent SRV_Channels changes

thanks to Jacob for noticing this!

### CODE REVIEW

**Comment by jaxxzer** [comment]
Thanks @tridge lgtm, I like it.

File: libraries/AP_ServoRelayEvents/AP_ServoRelayEvents.cpp [hunk 1]
```diff
@@ -36,8 +36,11 @@ bool AP_ServoRelayEvents::do_set_servo(uint8_t _channel, uint16_t pwm)
         // cancel previous repeat
         repeat = 0;
     }
-    hal.rcout->enable_ch(_channel-1);
-    hal.rcout->write(_channel-1, pwm);
+    SRV_Channel *c = SRV_Channels::srv_channel(_channel-1);
+    if (c == nullptr) {
```
Comment (jaxxzer) [inline libraries/AP_ServoRelayEvents/AP_ServoRelayEvents.cpp]: I think we should also check to make sure the channel function is *k_none* here.

**Comment by jaxxzer** [comment]
This will do the trick.
```
diff --git a/libraries/AP_ServoRelayEvents/AP_ServoRelayEvents.cpp b/libraries/AP_ServoRelayEvents/AP_ServoRelayEvents.cpp
index 68d73ff..aeedda3 100644
--- a/libraries/AP_ServoRelayEvents/AP_ServoRelayEvents.cpp
+++ b/libraries/AP_ServoRelayEvents/AP_ServoRelayEvents.cpp
@@ -37,7 +37,7 @@ bool AP_ServoRelayEvents::do_set_servo(uint8_t _channel, uint16_t pwm)
         repeat = 0;
     }
     SRV_Channel *c = SRV_Channels::srv_channel(_channel-1);
-    if (c == nullptr) {
+    if (c == nullptr || c->get_function() != SRV_Channel::k_none) {
         return false;
     }
     c->set_output_pwm(pwm);
diff --git a/libraries/SRV_Channel/SRV_Channel.h b/libraries/SRV_Channel/SRV_Channel.h
index 14321b3..a76c1d7 100644
--- a/libraries/SRV_Channel/SRV_Channel.h
+++ b/libraries/SRV_Channel/SRV_Channel.h
@@ -150,6 +150,11 @@ public:
         return servo_trim;
     }
 
+    // get FUNCTION parameter
+    Aux_servo_function_t get_function(void) const {
+        return (Aux_servo_function_t)function.get();
+    }
+
 private:
     AP_Int16 servo_min;
     AP_Int16 servo_max;
```

**Comment by jaxxzer** [comment]
We need to add a check for do_repeat_servo in the same manner, as well.

File: libraries/AP_ServoRelayEvents/AP_ServoRelayEvents.cpp (hunk 1 above)
Comment (tridge) [inline libraries/AP_ServoRelayEvents/AP_ServoRelayEvents.cpp]: that's right. It is actually deliberate, and matches previous behaviour. For example, you can DO_SET_SERVO on a bottle drop function channel.

**Pull request merged**